src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -178,9 +184,10 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
       return
     }
 
-    const { system_id, component_id } = mavlink_message.header
+    const { component_id } = mavlink_message.header
+    this.currentSystemId = mavlink_message.header.system_id
```

Comment:
This is not needed, and can cause troubles in case we receive messages from other vehicles (think of vehicles connected through mavlink endpoints), so we should remove it.